src/keys-builder/template/structural-directive.extractor.ts

```diff
@@ -76,34 +77,30 @@ export function traverse(
   }
 }
 
-/**
- * extract method calls from an AST.
- *
- * @example
- * {{ t('another(test)') | pipe1 | pipe2 }} will return the `t('another(test)')` node.
- *
- * @example
- * {{ data | pipe: t('key') }} will return the `t('key')` node.
- */
-function unwrapExpression(exp: AST): AST[] {
-  if (isBindingPipe(exp)) {
-    return [
-      ...unwrapExpression(exp.exp),
-      ...exp.args.map((arg) => unwrapExpression(arg)).flat(),
-    ];
-  } else if (isLiteralMap(exp)) {
-    return exp.values.map((value) => unwrapExpression(value)).flat();
+class MethodCallUnwrapper extends RecursiveAstVisitor {
+  public expressions: MethodCall[] = [];
+
+  override visitMethodCall(method: MethodCall, context: any) {
+    this.expressions.push(method);
+    super.visitMethodCall(method, context);
   }
+}
```

Comment:
A couple of things:
1. You can remove the `public` keyword.
2. I don't see that the [RecursiveAstVisitor](https://github.com/angular/angular/blob/ea61ec25628206d18a424906f685c0d0fd6aa714/packages/compiler/src/expression_parser/ast.ts#L452) has a `visitMethodCall` method, am I missing something?
3. According to the docs there seems like the `visit` method should be overridden.

Reply:
`visitMethodCall` is for angular compiler v12 
in v13 that method has been refactored into `visitCall`
https://github.com/angular/angular/pull/42882

Reply:
>Classes that extend RecursiveAstVisitor should override this function to selectively visit the specified node

As I understand it, `visit` should be overriden to decide if the given node shoud be visited. In this case we cant to visit every nodes of the AST thus we do not alter `visit`.

IMO it is better to execute actions based on the type of node in the dedicated methods (`visitUnary`, `visitBinary`, ...) and not in `visit`.

Look at that implementation it's very similar to what I've written.
https://github.com/angular/angular/blob/8a99ef18b4ae58736f0b7457e88908d3adcad613/packages/compiler/src/expression_parser/parser.ts#L1371-L1377

Reply:
I think we'll wait with the upgrade to v13 thanks for the clarification.